### PR TITLE
[read-fonts] Add test case for SBIX overflow

### PR DESCRIPTION
**Pull request by drott**
Add test case from crbug.com/347835680 which triggered overflows in font data reading, which are fixed after #966.

Test for #959. 

Test fails only on 32 bit, for example building and testing with: `--target=i686-unknown-linux-gnu`:
`cargo test --target=i686-unknown-linux-gnu "sbix_strikes_count_overflow"`

### CODE REVIEW

**Comment by cmyr** [comment]
Shouldn't this be failing right now, since it doesn't include #966?

**Comment by drott** [comment]
> edit: oh, will this only overflow on a 32-bit arch?

Yes, should have mentioned that here, too.
`cargo test --target=i686-unknown-linux-gnu "sbix_strikes_count_overflow"`
as I mentioned in #959.

**Review by rsheeter — APPROVED** [review]

File: read-fonts/src/tables/sbix.rs [hunk 1]
```diff
@@ -22,3 +22,41 @@ impl<'a> Strike<'a> {
...
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{FontRef, TableProvider};
+
+    #[test]
+    fn sbix_strikes_count_overflow() {
+        // Contains an invalid `num_strikes` values which would move the cursor outside the able.
+        let test_case = &[
```
Comment (cmyr) [inline read-fonts/src/tables/sbix.rs:33]: just for informational purposes, there is a `test_helpers` module in this crate that contains a `BeBuffer` type that is intended for creating intelligible test cases for this sort of scenario; using that we could write a version of this that looked like,

```rust
#[test]
fn sbix_strikes_count_overflow() {
    // Contains an invalid `num_strikes` values which would move the cursor outside the able.
    let sbix = BeBuffer::new()
        .push(1u16) // version
        .push(0u16) // flags
        .push(u32::MAX);

    let table = Sbix::read(sbix.font_data(), 5);
    table.unwrap();
}
```

which makes it easier for the reader to understand the problem.

(I haven't checked that this example actually fails)

**Review by cmyr — APPROVED** [review]
mildly annoying that this doesn't currently fail in CI, but agree it's worth capturing and we can try to figure out an appropriate CI runner later.

File: read-fonts/src/tables/sbix.rs (hunk 1 above)
Comment (drott) [reply read-fonts/src/tables/sbix.rs]: Thanks for the suggestion and distilling the test case - indeed, this works (same, on 32-bit) only, but is indeed a smaller and more legible way to express that. I was expecting to run into trouble trying to make other tables align with sbix for this to trigger but using the `read_with_args` entry point makes this easier, thanks for demoing that.

**Comment by drott** [comment]
I am going to go ahead and merge the test case in the form that Colin suggested, more compact. Hope that is not a too generous interpretation of your LGTMs. Otherwise please speak up.
